/*
 * Copyright (c) 2009, 2011, Oracle and/or its affiliates. All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice (including the next
 * paragraph) shall be included in all copies or substantial portions of the
 * Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 */

---
 driver/lock.c         |    4 ++--
 driver/xscreensaver.c |    1 -
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git driver/lock.c driver/lock.c
--- driver/lock.c
+++ driver/lock.c
@@ -918,6 +918,7 @@ new_passwd_window (saver_info *si)
                blurb(), pw->prompt_screen->number);
   }
 
+#ifdef CR7033508
   /* Before mapping the window, save a pixmap of the current screen.
      When we lower the window, we restore these bits.  This works,
      because the running screenhack has already been sent SIGSTOP, so
@@ -939,6 +940,7 @@ new_passwd_window (saver_info *si)
 	       0, 0);
     XFreeGC (si->dpy, gc);
   }
+#endif /*CR7033508*/
 
   si->pw_data = pw;
   return 0;
@@ -3102,8 +3104,6 @@ unlock_p (saver_info *si)
       return False;
     }
 
-  raise_window (si, True, True, True);
-
   xss_authenticate(si, p->verbose_p);
 
   return (si->unlock_state == ul_success);
diff --git driver/xscreensaver.c driver/xscreensaver.c
--- driver/xscreensaver.c
+++ driver/xscreensaver.c
@@ -1283,7 +1283,6 @@ main_loop (saver_info *si)
       for (i = 0; i < si->nscreens; i++)
         kill_screenhack (&si->screens[i]);
 
-      raise_window (si, True, True, False);
       if (si->throttled_p || getuid () == 0)
         fprintf (stderr, "%s: not launching hack (throttled.)\n", blurb());
       else
